controllers/resource/resource_generator.go

```diff
@@ -38,6 +43,21 @@ const (
 const doguPodNamespace = "POD_NAMESPACE"
 const doguPodName = "POD_NAME"
 
+const operatorVolumeClientName = "k8s-dogu-operator"
+
+type volumeClientParamType string
+
+const configMapParamType volumeClientParamType = "configmap"
+
+type volumeClientParams struct {
+	Type    volumeClientParamType
+	Content interface{}
+}
+
+type volumeClientConfigMapContent struct {
+	Name string
+}
+
```

Comment:
We should move the handling of volume clients into a separate file for better understanding and cleanness.